File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.py
class Solution(object):
    
    def dis(p):
        return p[0]**2 + P[1]**2
    
    def kClosest(self, points, k):
        """
        :type points: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        return sorted(points, key = lambda x: x[0]**2 + x[1]**2)[:k]
        
        

File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.cpp
class Solution {
public:
    
    inline int dist(vector<int>& p) {
        return p[0]*p[0] + p[1]*p[1];
    }
    
    
    vector<vector<int>> kClosest(vector<vector<int>>& ps, int k) {
    
        priority_queue<pair<int, int>> maxPQ;
        
        for (int i = 0 ; i < k; i++) {
            pair<int, int> ent = {dist(ps[i]), i};
            maxPQ.push(ent);
        }
        
        for(int i = k; i < ps.size(); i++) {
            pair<int, int> ent = {dist(ps[i]), i};
            if (ent.first < maxPQ.top().first) {
                maxPQ.pop();
                maxPQ.push(ent);
            }
        }
        
        vector<vector<int>> answer;
        
        while (!maxPQ.empty()) {
            int entryIndex = maxPQ.top().second;
            maxPQ.pop();
            answer.push_back(ps[entryIndex]);
        }
        
        return answer;
        
        
        
        
        
        
    }
};